winml/lib/Api/LearningModelSession.cpp

```diff
@@ -111,8 +111,20 @@ void LearningModelSession::Initialize() {
   }
 
   // Make onnxruntime apply the batch size override, if any
-  if (session_options_ && session_options_.BatchSizeOverride() != 0) {
-    WINML_THROW_IF_FAILED(engine_builder->SetBatchSizeOverride(session_options_.BatchSizeOverride()));
+  if (session_options_) {
+    if (session_options_.BatchSizeOverride() != 0) {
+      WINML_THROW_IF_FAILED(engine_builder->SetBatchSizeOverride(session_options_.BatchSizeOverride()));
+    }
+    // Make Onnxruntime apply the number of intra op threads
+    uint32_t numIntraOpThreads = session_options_.as<WINMLP::LearningModelSessionOptions>()->GetIntraOpNumThreads();
+    WINML_THROW_IF_FAILED(engine_builder->SetIntraOpNumThreadsOverride(numIntraOpThreads)
+    );
+  } else {
+    // Onnxruntime will use half the number of concurrent threads supported on the system
+    // by default. This causes MLAS to not exercise every logical core.
+    // If session options aren't provided, force the thread pool size to be maxxed out
+    // to ensure that WinML always runs the fastest.
+    WINML_THROW_IF_FAILED(engine_builder->SetIntraOpNumThreadsOverride(std::thread::hardware_concurrency()));
```

Comment:
Does it have to be set here and in https://github.com/microsoft/onnxruntime/pull/4638/files#diff-0e8f6e1e494e6761e670ef48ef82421eR53 ?

Reply:
Yes because LearningModelSessionOptions can be used for other use cases such as BatchSizeOverride without setting the number of IntraOp threads, there needs to be a default value to the member variable.